src/coordinates.rs

```diff
@@ -0,0 +1,164 @@
+//! Custom data types used in the Gerber format.
+
+use std::convert::{From, Into};
+
+use ::GerberError;
+
+
+/// The coordinate format specifies the number of integer and decimal places in
+/// a coordinate number. For example, the `24` format specifies 2 integer and 4
+/// decimal places. The number of decimal places must be 4, 5 or 6. The number
+/// of integer places must be not more than 6. Thus the longest representable
+/// coordinate number is `nnnnnn.nnnnnn`.
+#[derive(Debug, Copy, Clone)]
+pub struct CoordinateFormat {
+    pub integer: u8,
+    pub decimal: u8,
+}
+
+impl CoordinateFormat {
+    pub fn new(integer: u8, decimal: u8) -> Self {
+        CoordinateFormat { integer: integer, decimal: decimal }
+    }
+}
+
+
+/// Coordinate numbers are integers conforming to the rules set by the FS
+/// command.
+///
+/// Coordinate numbers are integers. Explicit decimal points are not allowed.
+///
+/// A coordinate number must have at least one character. Zero therefore must
+/// be encoded as `0`.
+///
+/// The value is stored as a 64 bit integer with 6 decimal places.
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub struct CoordinateNumber {
+    nano: i64,
+}
+
+const DECIMAL_PLACES_CHARS: u8 = 6;
+const DECIMAL_PLACES: i64 = 1_000_000;
+
+impl From<f64> for CoordinateNumber {
+    fn from(val: f64) -> CoordinateNumber {
+        CoordinateNumber { nano: (val * DECIMAL_PLACES as f64) as i64 }
```

Comment:
Jep, in debug mode it panics, but in release mode the integers wrap: https://is.gd/1jkKi7

We should probably use `checked_mul` instead: https://doc.rust-lang.org/nightly/std/primitive.i64.html#method.checked_mul